### DISCUSSION TITLE
Codespaces: Please allow VSCode users to choose between local port forwarding and githubpreview.dev forwarding

### DISCUSSION THREAD

**Discussion by ZhangY-MSFT** [community/community#3944] [Codespaces] [answered]
I noticed that recently Codespaces started to forward HTTP/HTTPS protocols with githubpreview.dev proxy - which I assume could allow sharing better, while also provide a solution to users of browser-based clients.

However, there are scenarios that I would like to keep direct port forwarding to localhost, and it feels a little bit concerning that I am not even allowed to choose between local port forwarding and githubpreview.dev, once HTTP/HTTPS is detected.

It is totally fine to use the proxy by default as it should cover the majority of use cases, but I think it might be a better idea to provide an option to keep local port forwarding for VSCode users.

**Comment by 2percentsilk** [comment]
Are you using the browser or local VS Code installation to use Codespaces?

And what are the scenarios where you'd like to keep port forwarding mapped to localhost?
Is there something in particular that makes you concerned about the githubpreview.dev URLS?

**Reply by ZhangY-MSFT** [reply]
The simplest one is for those that want to do a test server that interacts with other API, but cannot do it using private settings (due to auth and CORS), but are also not comfortable making their test servers public to the whole Internet.

**Comment by ZhangY-MSFT** [comment, marked as answer]
I found the switch Github › Codespaces: Use Port Forwarding Service For HTTPS that does the trick for my case. It'd be better to have a per-port choice in the ports menu, but it is no longer a blocker at least for me.
